### PR TITLE
Support 3/4/5 dimensional input for bias_add with NCHW data format

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address part of the issue raised in #20527 where biad_add only support 4 dimensional input with NCHW data format, and is causing tf.layers.conv3d to not suppport dynamic shapes with `channel_first` data format.

This fix add the 3/4/5 dimensional input for bias_add with NCHW data format as the first step to address the issue. Follow up PR will be added to fix tf.layers.conv3d issue.

This fix also only adds bias_add support. BiasGradOp still only works with 4D+NCHW. The 3D/5D+NCHW+BiadGradOp will be worked on later (TODO).

This fix is part of the effort for #20527.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by frankchn** [comment]
Hi @yongtang Can you fix the test and try again? The code looks good otherwise.

```tensorflow/core/framework/common_shape_fns_test.cc:374
Expected equality of these values:
  3
  c.Value(c.Dim(output, 0))
    Which is: 6```

**Review by frankchn — DISMISSED** [review]

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yongtang** [comment]
@yifeif The PR has been rebased to resolve the merge conflict.

**Comment by frankchn** [comment]
This looks like it might be breaking biasadd_matmul_test.py, since TensorRT has a different idea of what NCHW means when it comes to 5-dimension and 3-dimension Tensors. I am not sure what is the best way forward here. 

cc @jjsjann123, @aaroey, @yifeif

**Comment by frankchn** [comment]
@aaroey fixed the test internally, so we are good. This should go in shortly...
